core/orchestrator.go

```diff
@@ -150,12 +150,14 @@ func (orch *orchestrator) ProcessPayment(payment net.Payment, manifestID Manifes
 
 	for _, tsp := range payment.TicketSenderParams {
 
+		// shadow ticket declaration & allocate new instance to prevent SenderNonce race conditions
+		ticket := *ticket
```

Comment:
A more explicit (and IMO a more readable) approach that doesn't require copying all the ticket fields (though technically, only the struct fields that are values are copied and not the struct fields that are pointers)  in each iteration is to use the `pm.NewTicket()` constructor within the loop.

Ex.

```
ticketParams := &pm.TicketParams{
    // Initialize using payment.TicketParams
}

expirationParams := &pm.TicketExpirationParams{
    // Initialize using payment.ExpirationParams
}

for _, tsp := range payment.TicketSenderParams {
    ticket := pm.NewTicket(ticketParams, expirationParams, sender, tsp.SenderNonce)
    ...
}
```

In the above code snippet, we can create the `ticketParams` and `expirationParams` pointers once and then each iteration of the loop will create a ticket that references the same `ticketParams` and `expirationParams`, but uses a different `senderNonce`.

Something to make note of is that if multiple goroutines are created by the loop then a data race could be possible if we ever modify the `ticketParams` and/or `expirationParams` pointers. There is a similar concern if we took the approach of copying the ticket in each iteration of the loop except the data race could be around `ticket.FaceValue` and `ticket.WinProb` since those are the pointers shared by each ticket copy. One way to completely eliminate the possibility of data races is to create a fresh ticket in each iteration of the loop that does not share *any* pointers with other tickets. But, at the moment since we're never modifying `ticketParams` or `expirationParams` after creating them, we should be ok.

Reply:
Is a9075d4 this something you were thinking of in terms of testing or how to better approach this case?

Reply:
I was just highlighting something that I think is worth being aware of. I don't think we need to take further action for now.